src/types/base.ts

```diff
@@ -59,12 +59,13 @@ export interface Metric {
   entries: (PerformanceEntry | LayoutShift | FirstInputPolyfillEntry | NavigationTimingPolyfillEntry)[];
 
   /**
-   * For regular navigations, the type will be the same as the type indicated
-   * by the Navigation Timing API (or `undefined` if the browser doesn't
+   * The type of navigation
+   *
+   * Navigation Timing API (or `undefined` if the browser doesn't
    * support that API). For pages that are restored from the bfcache, this
-   * value will be 'back_forward_cache'.
+   * value will be 'back-forward-cache'.
    */
-  navigationType:  NavigationTimingType | 'back_forward_cache' | 'prerender' | undefined;
+  navigationType:  'navigate' | 'reload' | 'back-forward' | 'back-forward-cache' | 'prerender';
```

Comment:
On line 64 you mention `undefined` but have remove that from here?

Reply:
Correct, I updated the code so the `undefined` would no longer be possible.

Note, I don't think it was ever technically possible before, but according to the type logic is was possible since `getNavigationEntry()` can return `undefined` in cases where the browser doesn't support any performance APIs at all (see #185). However, browsers that do not support any performance APIs should not be able to get this far in the code to ever fire these callbacks (all `onXXX()` functions are gated on _some_ sort of performance API support), which is why I say I don't think it's technically possible, at least not in any  browser I'm aware of.

I made the change because I thought it was much riskier to have the type definition advertise that it is possible to get `undefined` here, and I didn't want developers to be coding defensively for something that I don't think can happen (and now definitely can't because the default is `'navigate'`).